projects/ngx-datatable/src/lib/components/header/header-cell.component.ts

```diff
@@ -295,40 +298,16 @@ export class DataTableHeaderCellComponent implements OnInit, OnDestroy {
     }
   }
 
-  protected onMousedown(event: MouseEvent | TouchEvent): void {
-    const isMouse = event instanceof MouseEvent;
-    const initialWidth = this.element.clientWidth;
-    const { screenX } = getPositionFromEvent(event);
-    event.stopPropagation();
-
-    const mouseup = fromEvent<MouseEvent | TouchEvent>(document, isMouse ? 'mouseup' : 'touchend');
-    this.subscription = mouseup.subscribe(() => this.onMouseup());
-
-    const mouseMoveSub = fromEvent<MouseEvent | TouchEvent>(
-      document,
-      isMouse ? 'mousemove' : 'touchmove'
-    )
-      .pipe(takeUntil(mouseup))
-      .subscribe((e: MouseEvent | TouchEvent) => this.move(e, initialWidth, screenX));
-
-    this.subscription.add(mouseMoveSub);
+  protected onMousedown(): void {
+    this.initialWidth = this.element.clientWidth;
   }
 
-  private onMouseup(): void {
-    if (this.subscription && !this.subscription.closed) {
-      this.destroySubscription();
-      this.resize.emit({ width: this.element.clientWidth, column: this.column });
-    }
+  protected onMouseup(): void {
+    this.resize.emit({ width: this.element.clientWidth, column: this.column });
   }
 
-  private move(
-    event: MouseEvent | TouchEvent,
-    initialWidth: number,
-    mouseDownScreenX: number
-  ): void {
-    const movementX = getPositionFromEvent(event).screenX - mouseDownScreenX;
-    const newWidth = initialWidth + movementX;
-    this.resizing.emit({ width: newWidth, column: this.column });
+  protected move({ currentX, initialX }: DragEvent): void {
+    this.resizing.emit({ width: this.initialWidth! + (currentX - initialX), column: this.column });
```

Comment:
let's not assume that initialWidth will always have value
```suggestion
if(this.initialWidth !== undefined) {
    this.resizing.emit({ width: this.initialWidth + (currentX - initialX), column: this.column });
    }
```